vitrina/messages/models.py

```diff
@@ -161,3 +111,37 @@ def get_absolute_url(self):
             return self.content_object.get_absolute_url()
         else:
             return None
+
+
+class NewsletterSubscriber(models.Model):
+    email = models.EmailField(unique=True)
+    created = models.DateTimeField(auto_now_add=True)
+    is_confirmed = models.BooleanField(default=False)
+    is_active = models.BooleanField(default=True)
+
+    confirmation_token = models.UUIDField(
+        default=uuid.uuid4, unique=True, null=True, blank=True
+    )
+    confirmation_expires_at = models.DateTimeField(null=True, blank=True)
+
+    unsubscribe_token = models.UUIDField(default=uuid.uuid4, unique=True)
+
+    def save(self, *args, **kwargs):
+        if not self.is_confirmed and not self.confirmation_expires_at:
+            self.confirmation_expires_at = timezone.now() + timedelta(hours=24)
+        super().save(*args, **kwargs)
+
+    def is_confirmation_expired(self):
```

Comment:
čia galima būtų pažymėti su `Čproperty` dekoratorium ir po to naudoti kaip property